kubernetes/template-generator/template_generator.py

```diff
@@ -0,0 +1,359 @@
+            @param ContainerOS system       a os description module
+            @param list exposed_ports       a list of exposed ports
+            @param Boolean is_local         disable pulling the image from registry on node
+        """
+        super(MacroimageTemplateGenerator, self).__init__(container_name, system, exposed_ports, external_ips)
+
+        self.is_local = is_local
+
+    def macroimage_name(self):
+        """
+            generate docker image name
+        """
+        return "leapp/{}".format(self.container_name)
+
+    def generate_pod_template(self, yaml=True):
+        """
+            Generate a YAML template for kerberos pod
+            @param Boolean yaml     if true, the output will be in yaml, else a
+                                    dict will be returned
+
+            @return string          a template string in yaml format/dict
+        """
+        pod_template = TemplateGenerator.generate_pod_template(self, False)
+
+        pod_template["spec"]["containers"][0]["image"] = self.macroimage_name()
+
+        if self.is_local:
+            pod_template["spec"]["containers"][0]["imagePullPolicy"] = "Never"
+
+        if yaml:
+            return self._dump_yaml(pod_template)
+
+        return pod_template
+
+    def generate_dockerfile(self):
+        """
+            Method generate Dockerfile for macroimage. The dockerfile expects the tar file
+            in the same directory (scope) names as: sanitized-container-name.tar.gz
+
+            @return string  a dockerfile content
+        """
+        # TODO: This can be done better if the image is being build on the same machine
+        #       where the leapp tool is running
+        path = "{}.tar.gz".format(self.container_name)
+        content = "FROM {0}\nADD {1} /".format(self.system.base_image(), path)
+
+        return content
+
+
+class InvalidVolumeMount(ValueError):
+    pass
+
+
+class ForbiddenVolumeMount(ValueError):
+    pass
+
+
+class MultivolumeTemplateGenerator(TemplateGenerator):
+    FORBIDDEN_MOUNTS = ["dev", "sys", "proc", "lost+found"]
+
+    def __init__(self, container_name, system, image_url, exposed_ports=None, external_ips=None,
+                 exported_paths=None):
+        """
+            @param string container_name    the name of migrated container
+            @param ContainerOS system       a os description module
+            @param string image_url         an URL to tar image containing the container FS
+            @param list exported_paths      a list of all paths for which should be created a volume
+            @param list exposed_ports       a list of exposed ports
+
+            @raise InvalidVolumeMount
+            @raise ForbiddenVolumeMount
+        """
+        super(MultivolumeTemplateGenerator, self).__init__(container_name, system, exposed_ports, external_ips)
+
+        if not exported_paths:
+            raise AttributeError("Exported paths has to be set")
```

Comment:
This should be `ValueError` as `AttributeError` is raised when you try to access invalid attribute like `object.attr`.